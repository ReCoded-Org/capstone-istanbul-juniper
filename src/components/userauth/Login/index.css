.loginContainer__loginTitle {
  font-family: "Salsa";
}
