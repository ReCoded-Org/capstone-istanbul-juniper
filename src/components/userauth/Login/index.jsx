import { Alert, Input, Button } from "antd";
import React, { useState } from "react";
import { useTranslation } from "react-i18next";
import { validateEmail } from "../registerHelper";

const Login = ({
  // check containers/LoginRegister/index.jsx to see details of functions
  onSubmit,
  handleFacebookAuth,
  handleGoogleAuth,
  handleRegister,
  handlePasswordReset,
  // string
  error,
  // string
  message,
}) => {
  const [t] = useTranslation();
  const [loginCredentials, setLoginCredentials] = useState({
    email: "",
    password: "",
  });
  const handleChange = (key, value) => {
    const newValues = { ...loginCredentials };
    newValues[key] = value;
    setLoginCredentials(newValues);
  };
  const [errors, setErrors] = useState({});
  const handleSubmit = (e) => {
    e.preventDefault();
    const { email, password } = loginCredentials;
    const newErrors = {};
    if (!validateEmail(email)) {
      newErrors.email = "Email format error";
    }
    if (email.trim() === "") {
      newErrors.email = "Please fill this field";
    }
    if (password.trim().length < 6) {
      newErrors.password = "Password must be at least 6 characters long";
    }
    if (password.trim() === "") {
      newErrors.password = "Please fill this field";
    }
    setErrors(newErrors);
    const errorNameArr = Object.keys(newErrors);
    if (errorNameArr.length > 0) {
      return;
    }
    onSubmit(loginCredentials);
  };

  return (
    <form noValidate onSubmit={handleSubmit}>
      <div className="loginContainer">
        <div className="loginContainer__loginTitle">{t("login.login")}</div>
        {error && (
          <Alert
            style={{ marginBottom: 10 }}
            type="error"
            showIcon
            message={error}
          />
        )}
        {message}
        <div className="loginContainer__loginDialog">
          <div className="loginContainer__loginDialog__inputLabel">
            {t("login.email")}
          </div>
          <div className="loginContainer__loginDialog__input">
            <Input
              className={
                errors.email && "loginContainer__loginDialog__input__hasError"
              }
              type="email"
              value={loginCredentials.email}
              onChange={(e) => {
                handleChange("email", e.target.value);
              }}
            />
            {errors.email && (
              <div className="loginContainer__loginDialog__errorContainer">
                {errors.email}
              </div>
            )}
          </div>
          <div className="loginContainer__loginDialog__inputLabel">
            {t("login.password")}
          </div>
          <div className="loginContainer__loginDialog__input">
            <Input
              className={
                errors.password &&
                "loginContainer__loginDialog__input__hasError"
              }
              type="password"
              value={loginCredentials.password}
              onChange={(e) => {
                handleChange("password", e.target.value);
              }}
            />
            {errors.password && (
              <div className="loginContainer__loginDialog__errorContainer">
                {errors.password}
              </div>
            )}
          </div>
          <div className="loginContainer__loginDialog__forgotPasswordContainer">
            <Button
              type="link"
              onClick={(e) => {
                e.preventDefault();
                handlePasswordReset();
              }}
            >
              {t("login.forgotPassword")}
            </Button>
          </div>
          <button
            type="submit"
            className="loginContainer__loginDialog__submitButton"
          >
            {t("login.login")}
          </button>
          <div className="loginContainer__loginDialog__registerLabel">
            {t("login.new")}
          </div>
          <button
            onClick={handleRegister}
            type="button"
            className="loginContainer__loginDialog__registerButton"
          >
            {t("login.creat")}
          </button>
          <div className="loginContainer__loginDialog__loginOptionsTitle">
            {t("login.oryoucan")}
          </div>
          <button
            onClick={handleFacebookAuth}
            type="button"
            className="loginContainer__loginDialog__facebookLoginBtn"
          >
            {t("login.withFacebook")}
          </button>
          <button
            onClick={handleGoogleAuth}
            type="button"
            className="loginContainer__loginDialog__googleLoginBtn"
          >
            {t("login.withGoogle")}
          </button>
        </div>
      </div>
    </form>
  );
};

export default Login;
