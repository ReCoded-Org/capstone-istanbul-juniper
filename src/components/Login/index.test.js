import React from "react";
import Login from "./index.jsx";
import renderer from "react-test-renderer";

describe("Login Component", () => {
  it("matches the tree", () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
