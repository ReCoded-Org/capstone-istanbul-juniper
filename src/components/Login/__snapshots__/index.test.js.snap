// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component matches the tree 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="loginContainer"
  >
    <div
      className="loginContainer__loginTitle"
    >
      login.login
    </div>
    <div
      className="ant-alert ant-alert-error"
      data-show={true}
      role="alert"
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <span
        aria-label="close-circle"
        className="anticon anticon-close-circle ant-alert-icon"
        role="img"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="close-circle"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
          />
        </svg>
      </span>
      <span
        className="ant-alert-message"
      />
      <span
        className="ant-alert-description"
      />
    </div>
    <div
      className="loginContainer__loginDialog"
    >
      <div
        className="loginContainer__loginDialog__inputLabel"
      >
        login.email
      </div>
      <div
        className="loginContainer__loginDialog__input"
      >
        <input
          className="ant-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          type="email"
          value=""
        />
      </div>
      <div
        className="loginContainer__loginDialog__inputLabel"
      >
        login.password
      </div>
      <div
        className="loginContainer__loginDialog__input"
      >
        <input
          className="ant-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          type="password"
          value=""
        />
      </div>
      <div
        className="loginContainer__loginDialog__forgotPasswordContainer"
      >
        <a
          href="/#"
          onClick={[Function]}
        >
          login.forgotPassword
        </a>
      </div>
      <button
        className="loginContainer__loginDialog__submitButton"
        type="submit"
      >
        login.login
      </button>
      <div
        className="loginContainer__loginDialog__registerLabel"
      >
        login.new
      </div>
      <button
        className="loginContainer__loginDialog__registerButton"
        onClick={[Function]}
        type="button"
      >
        login.creat
      </button>
      <div
        className="loginContainer__loginDialog__loginOptionsTitle"
      >
        login.oryoucan
      </div>
      <button
        className="loginContainer__loginDialog__facebookLoginBtn"
        onClick={[Function]}
        type="button"
      >
        login.withFacebook
      </button>
      <button
        className="loginContainer__loginDialog__googleLoginBtn"
        onClick={[Function]}
        type="button"
      >
        login.withGoogle
      </button>
    </div>
  </div>
</form>
`;
